src/backend/distributed/connection/remote_commands.c

```diff
@@ -326,6 +328,38 @@ SendRemoteCommand(MultiConnection *connection, const char *command)
 
 
 /*
+ * ReadRemoteQueryResults reads first columns of result tuples from the given
+ * PGresult struct and returns them in a StringInfo list.
+ */
+List *
+ReadRemoteQueryResults(PGresult *queryResult)
+{
+	List *resultRowList = NIL;
+	const int columnIndex = 0;
+	int64 rowIndex = 0;
+	int64 rowCount = 0;
+
+	ExecStatusType status = PQresultStatus(queryResult);
+	if (status == PGRES_TUPLES_OK)
+	{
+		rowCount = PQntuples(queryResult);
+	}
+
+	for (rowIndex = 0; rowIndex < rowCount; rowIndex++)
+	{
+		char *rowValue = MultiClientGetValue(queryResult, rowIndex, columnIndex);
```

Comment:
This is just a thin (deprecated) wrapper around PQgetvalue, so maybe just call that directly.